poppy/poppy_core.py

```diff
@@ -437,16 +432,16 @@ def display(self,what='intensity', nrows=1,row=1,showpadding=False,imagecrop=Non
                 title = "Intensity "+self.location
                 title = title.replace('after', 'after\n')
                 title = title.replace('before', 'before\n')
-            plt.title(title)
-            plt.xlabel(unit)
+            ax.set_title(title)
+            ax.set_xlabel(unit)
             if colorbar: plt.colorbar(ax.images[0], orientation='vertical', shrink=0.8)
 
-            if self.planetype ==_IMAGE:
+            if is_angular:
```

Comment:
Would it be possible for FresnelWavefront to accept the desired units in a subclassed version of `Wavefront.display`? If it's only used for plotting, I'm leery of adding more state to the wavefront